src/game_inventory.cpp

```diff
@@ -1828,13 +1836,30 @@ class bionic_install_preset: public inventory_selector_preset
             int chance_of_failure = 100;
             player &installer = p;
 
+            const std::string bionic_name = loc.get_item()->typeId().c_str();
+
+            std::vector<const item *> install_programs = p.crafting_inventory().items_with( [](
+                        const item & it ) -> bool { return it.has_flag( flag_BIONIC_INSTALLATION_DATA ); } );
+
+            bool has_install_program = false;
+
+            if( !install_programs.empty() ) {
+                for( const item *progs : install_programs ) {
+                    const std::string AID_name = progs->typeId().c_str();
+                    has_install_program = AID_name.substr( 4 ) == bionic_name;
```

Comment:
Sorry to comment after this has been merged, but I don't think it's good to make this functionality based on the ids of the items.
I don't think string hacking with the ids is good practice, and it places restrictions on how you can use these.
If you didn't implement it this way, and instead had some sort of field, this would be much more flexible, and would also very easily allow extension of this, such as the original implementation of it.

e.g. There's rare items that ease a large class of CBM installs, and less rare variants allowing only one or a few.

Reply:
I just went the easiest IMO way to implement it. The stuff can be refactored later if you think my approach is _so_ bad.